moveit_ros/move_group/src/default_capabilities/query_planners_service_capability.cpp

```diff
@@ -58,23 +58,35 @@ void MoveGroupQueryPlannersService::initialize()
 bool MoveGroupQueryPlannersService::queryInterface(moveit_msgs::QueryPlannerInterfaces::Request& /*req*/,
                                                    moveit_msgs::QueryPlannerInterfaces::Response& res)
 {
-  const planning_interface::PlannerManagerPtr& planner_interface = context_->planning_pipeline_->getPlannerManager();
-  if (planner_interface)
+  for (const auto& planning_pipelines : context_->moveit_cpp_->getPlanningPipelines())
   {
-    std::vector<std::string> algs;
-    planner_interface->getPlanningAlgorithms(algs);
-    moveit_msgs::PlannerInterfaceDescription pi_desc;
-    pi_desc.name = planner_interface->getDescription();
-    planner_interface->getPlanningAlgorithms(pi_desc.planner_ids);
-    res.planner_interfaces.push_back(pi_desc);
+    const auto& pipeline_id = planning_pipelines.first;
+    const auto& planning_pipeline = planning_pipelines.second;
+    const planning_interface::PlannerManagerPtr& planner_interface = planning_pipeline->getPlannerManager();
+    if (planner_interface)
+    {
+      std::vector<std::string> algs;
+      planner_interface->getPlanningAlgorithms(algs);
+      moveit_msgs::PlannerInterfaceDescription pi_desc;
+      pi_desc.name = planner_interface->getDescription();
+      // Set pipeline id for non-default planners
+      if (planning_pipeline != context_->planning_pipeline_)
+        pi_desc.pipeline_id = pipeline_id;
```

Comment:
Why not setting the pipeline_id for the default planner as well?

Reply:
Originally, to support specifying the default and to keep consistent with old launch files. Both points don't really hold true anymore so I fixed it as suggested -> all pipelines are specified by their ids.